### PR TITLE
CIV-10058 caseworker for no remission hwf

### PR DESCRIPTION
**Pull request by jeswanth-hmcts**
**Before creating a pull request make sure that:**

- [x] commit messages are meaningful and follow good commit message guidelines
- [x] README and other documentation has been updated / added (if needed)
- [x] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/CIV-10058

### Change description ###
 caseworker for no remission hwf

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/civil/model/citizenui/HelpWithFeesDetails.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+@Data
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
+public class HelpWithFeesDetails {
+
+    private String noRemissionDetails;
+    private HWFFeeDetailsSummary noRemissionDetailsSummary;
+    private FeeType hwfFeeType;
+
```
Comment (manish14836) [inline src/main/java/uk/gov/hmcts/reform/civil/model/citizenui/HelpWithFeesDetails.java:19]: Remove extra line

File: src/test/java/uk/gov/hmcts/reform/civil/handler/callback/user/NoRemissionHWFCallbackHandlerTest.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+            CaseData caseData = CaseData.builder()
+                .hwFeesDetails(HelpWithFeesDetails.builder().noRemissionDetails("no remission")
+                                   .noRemissionDetailsSummary(HWFFeeDetailsSummary.FEES_REQUIREMENT_NOT_MET).hwfFeeType(
+                        FeeType.CLAIMISSUED).build())
+                .build();
+            CallbackParams params = callbackParamsOf(caseData, CallbackType.ABOUT_TO_SUBMIT);
+            //When
+            var response = (AboutToStartOrSubmitCallbackResponse) handler.handle(params);
+            //Then
+            assertThat(response.getData()).extracting("hwFeesDetails").isNotNull();
```
Comment (manish14836) [inline src/test/java/uk/gov/hmcts/reform/civil/handler/callback/user/NoRemissionHWFCallbackHandlerTest.java:49]: Add more specific asserts like comparing values you have set in caseData fields

File: src/main/java/uk/gov/hmcts/reform/civil/model/citizenui/HelpWithFeesDetails.java (hunk 1 above)
Comment (jeswanth-hmcts) [reply src/main/java/uk/gov/hmcts/reform/civil/model/citizenui/HelpWithFeesDetails.java]: removed the extra line

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/NoRemissionHWFCallbackHandler.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+@Service
+@RequiredArgsConstructor
+public class NoRemissionHWFCallbackHandler extends CallbackHandler {
+
+    private static final List<CaseEvent> EVENTS = List.of(NO_REMISSION_HWF);
+    private final ObjectMapper objectMapper;
+    private final Map<String, Callback> callbackMap = Map.of(
+        callbackKey(ABOUT_TO_SUBMIT),
+        this::noRemssionHWF,
```
Comment (portizdev) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/NoRemissionHWFCallbackHandler.java:28]: Typo error: noRmssion

File: src/main/java/uk/gov/hmcts/reform/civil/model/citizenui/HelpWithFeesDetails.java [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+import uk.gov.hmcts.reform.civil.enums.HWFFeeDetailsSummary;
+
+@Data
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
+public class HelpWithFeesDetails {
+
+    private String noRemissionDetails;
+    private HWFFeeDetailsSummary noRemissionDetailsSummary;
```
Comment (neeta-hmcts) [inline src/main/java/uk/gov/hmcts/reform/civil/model/citizenui/HelpWithFeesDetails.java:17]: HWFFeeDetailsSummary This enum class is specific to noRemission so can you change the name accordingly?

**Review by neeta-hmcts — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/civil/model/citizenui/HelpWithFeesDetails.java (hunk 4 above)
Comment (jeswanth-hmcts) [reply src/main/java/uk/gov/hmcts/reform/civil/model/citizenui/HelpWithFeesDetails.java]: changed to noRemissionDetailsSummary

**Review by portizdev — DISMISSED** [review]

**Review by ElifDenizEsea — APPROVED** [review]

**Review by sankhajuria — APPROVED** [review]

**Pull request merged**